content/realtime/channels/channel-parameters/deltas.textile

```diff
@@ -0,0 +1,207 @@
+---
+title: Delta compression
+section: realtime
+index: 3
+jump_to:
+  Help with:
+    - Overview#overview
+    - Delta processing#delta-processing
+    - Using deltas#using-deltas
+    - Examples#examples
+---
+
+h2(#overview). Overview
+
+The @delta@ parameter enables delta compression. It is applied on the channel you are subscribing to, enabling delta mode. 
```

Comment:
This documentation makes no reference to the fact that not all SDKs support this functionality.
In addition, is there a PR to address https://www.ably.io/download/sdk-feature-support-matrix?

Reply:
@tomczoink do we have a typical way of showing/stating what SDKs are supported? Also how do we handle content updates outside of the docs repo like the one Matt has mentioned?

Reply:
Example:

languages:
  - javascript
  - nodejs
  - ruby
  - java
  - swift
  - objc
  - csharp

Reply:
@tomczoink if a language is not included in the `languages` list, what happens? Are they taken to a different page?  To be honest, it's been a while so I can't recall.  I think we should:

* Established what happens when you specify languages in the list (I think this only affects which code blocks are shown)
* Document it in https://docs.ably.io/client-lib-development-guide/documentation-formatting-guide/#language-specific-content
* Think about how we show users that in their selected language, this feature is NOT supported, and link to the features matrix.

Reply:
Currently the behaviour on the website is that the language will not appear in the top bar, and any code snippets will state 'there is no example in this language'. They aren't redirected or anything. Currently we don't have any warnings on pages automatically generated for non-supported languages, we instead manually add in caveats as needed (for example https://www.ably.io/documentation/general/push#download).

Personally I'd be happy for us to have some banner which is created for non-supported languages being selected, perhaps with optionally customizable text supplied via metadata or something. Whether we'd also want all languages to appear in the top languages bar, just providing a warning when selected, should be considered.

I'd suggest for now we keep this sort of change seperate from this release, as there's still a lot of work to be done in this PR for the intended release of this week. I can create an issue where we could discuss this further and get a PR up from that once we're finished with deltas.

For this PR, I'd suggest we just place a basic notice at the top of the page which states that deltas is only supported by X, Y and Z.

Reply:
I think there's a misunderstanding about why that exists.

The explicit languages are not to add more languages, but instead to restrict.

For example, if you are showing a tutorial that is for Android and iOS, and it includes some HTML and Code blocks, it will automatically show Javascript in the nav because it detects all languages usde.  In most cases, that's what we want.  But we introduced that option ONLY to prevent languages no showing that you didn't want to show.